pkg/controller/kogitoinfra/infinispan/infinispan_reconciler.go

```diff
@@ -0,0 +1,133 @@
+		resourceNameSpace := instance.Spec.Resource.Namespace
+
+		infinispanInstance, resultErr = loadDeployedInfinispanInstance(client, resourceName, resourceNameSpace)
+		if resultErr != nil {
+			return false, resultErr
+		}
+		if infinispanInstance == nil {
+			return false, fmt.Errorf("custom Infinispan instance(%s) not found in namespace %s", resourceName, resourceNameSpace)
+		}
+	} else {
+		// create/refer kogito-infinispan instance
+		log.Debugf("Custom infinispan instance reference is not provided")
+		// if resource name is not provided then Infinispan instance should be created with default name = kogito-infinispan
+		resourceName := InstanceName
+
+		// if resource name is not provided then Infinispan instance should be created with default name = kogito-infinispan
+		resourceNameSpace := instance.Namespace
+
+		// Verify Infinispan
+		infinispanAvailable, err := infrastructure.IsInfinispanOperatorAvailable(client, resourceNameSpace)
+		if err != nil {
+			return false, err
+		}
+		if !infinispanAvailable {
+			err = fmt.Errorf("Infinispan operator is not available in the namespace %s, impossible to continue ", resourceNameSpace)
+			return false, err
+		}
+
+		// Step 1: check whether infinispan instance exist
+		infinispanInstance, resultErr = loadDeployedInfinispanInstance(client, resourceName, resourceNameSpace)
+		if resultErr != nil {
+			return false, resultErr
+		}
+
+		if infinispanInstance == nil {
+			// if not exist then create new Infinispan instance. Infinispan operator creates Infinispan instance, secret & service resource
+			infinispanInstance, resultErr = createNewInfinispanInstance(client, resourceName, resourceNameSpace, instance, scheme)
+			if resultErr != nil {
+				return false, resultErr
+			} // provisioning
+			return true, nil
+		}
+
+	}
+
+	customInfinispanSecret, resultErr := loadCustomKogitoInfinispanSecret(client, instance.Namespace)
+	if resultErr != nil {
+		return false, resultErr
+	}
+
+	if customInfinispanSecret == nil {
+		customInfinispanSecret, resultErr = createCustomKogitoInfinispanSecret(client, instance.Namespace, infinispanInstance, instance, scheme)
+		if resultErr != nil {
+			return false, resultErr
+		}
+		return true, nil
+	}
+
+	infinispanURI, resultErr := infrastructure.FetchKogitoInfinispanInstanceURI(client, infinispanInstance.Name, infinispanInstance.Namespace)
+	if resultErr != nil {
+		return false, resultErr
+	}
+
+	infinispanProperties := &instance.Status.Infinispan.InfinispanProperties
+	infinispanProperties.URI = infinispanURI
+	infinispanProperties.Credentials.SecretName = customInfinispanSecret.Name
+	infinispanProperties.Credentials.UsernameKey = infrastructure.InfinispanSecretUsernameKey
+	infinispanProperties.Credentials.PasswordKey = infrastructure.InfinispanSecretPasswordKey
+
+	log.Debugf("Updating kogitoInfra(%s) value with new properties", instance.Name)
+	if resultErr = kubernetes.ResourceC(client).Update(instance); resultErr != nil {
```

Comment:
Is this updating the status of the instance, right? Why we need this here? I didn't see a change in the `instance` attributes in this function.

Reply:
We are setting properties in instance attribute:
```
infinispanProperties := &instance.Status.Infinispan.InfinispanProperties
	infinispanProperties.URI = infinispanURI
	infinispanProperties.Credentials.SecretName = customInfinispanSecret.Name
	infinispanProperties.Credentials.UsernameKey = infrastructure.InfinispanSecretUsernameKey
	infinispanProperties.Credentials.PasswordKey = infrastructure.InfinispanSecretPasswordKey
```

Reply:
I had removed this update